src/App.js

```diff
@@ -1,23 +1,133 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+
+import Header from "./components/Header";
+import DisplayAlgo from "./components/DisplayAlgo";
+import DisplayAlgoLine from "./components/DisplayAlgoLine";
+import Footer from "./components/Footer";
+import { useState, useEffect } from "react";
 
 function App() {
+  const numRow = Math.floor(window.innerWidth / 11);
```

Comment:
A component in React is a function that gets executed every time there are changes to it's state (`useState`), or on it's received properties. This means the body of those components (aka functions) should avoid any potentially heavy operation, specially when these operations result in the same value.

To ensure you execute some calculation only once, you could rely on [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo):

```suggestion
  const numRow = useMemo(() => Math.floor(window.innerWidth / 11), []);
```

Even further, you could simplify it all by having this calculation happen _outside_ the function, meaning it would occur one time: when the code is first read/run:

```js

const numRow = Math.floor(window.innerWidth / 11);

function App() {
  // ...
}
```

> Keep in mind in this specific case here, we the calculation it relying on the `window.innerWidth`, which would change when a user resizes the browser's window. If we need to account for that, we would than better have this calculation be a state (`useState`), and rely on a `useEffect` to listen for window size changes to recalculate the value.